mmdet/apis/ote/apis/detection/nncf_task.py

```diff
@@ -0,0 +1,236 @@
+        self._should_stop = False
+
+    def _load_model(self, model: Model):
+        compression_ctrl = None
+        if model != NullModel():
+            # If a model has been trained and saved for the task already, create empty model and load weights here
+            buffer = io.BytesIO(model.get_data("weights.pth"))
+            model_data = torch.load(buffer, map_location=torch.device('cpu'))
+
+            model = self._create_model(self._config, from_scratch=True)
+            try:
+                if is_state_nncf(model_data):
+                    compression_ctrl, model = wrap_nncf_model(
+                        model,
+                        self._config,
+                        init_state_dict=model_data,
+                        get_fake_input_func=get_fake_input
+                    )
+                    logger.info("Loaded model weights from Task Environment and wrapped by NNCF")
+                else:
+                    try:
+                        model.load_state_dict(model_data['model'])
+                        logger.info(f"Loaded model weights from Task Environment")
+                        logger.info(f"Model architecture: {self._model_name}")
+                    except BaseException as ex:
+                        raise ValueError("Could not load the saved model. The model file structure is invalid.") \
+                            from ex
+
+                logger.info(f"Loaded model weights from Task Environment")
+                logger.info(f"Model architecture: {self._model_name}")
+            except BaseException as ex:
+                raise ValueError("Could not load the saved model. The model file structure is invalid.") \
+                    from ex
+        else:
+            raise ValueError(f"No trained model in project. NNCF require pretrained model to compress the model")
+        return compression_ctrl, model
+
+    def _create_compressed_model(self, dataset, config):
+        init_dataloader = build_dataloader(
+            dataset,
+            config.data.samples_per_gpu,
+            config.data.workers_per_gpu,
+            len(config.gpu_ids),
+            dist=False,
+            seed=config.seed)
+
+        self._compression_ctrl, self._model = wrap_nncf_model(
+            self._model,
+            config,
+            init_dataloader,
+            get_fake_input)
+
+    def optimize(
+        self,
+        optimization_type: OptimizationType,
+        dataset: Dataset,
+        output_model: Model,
+        optimization_parameters: Optional[OptimizationParameters],
+    ):
+        if optimization_type is not OptimizationType.NNCF:
+            raise RuntimeError("NNCF is the only supported optimization")
+
+        train_dataset = dataset.get_subset(Subset.TRAINING)
+        val_dataset = dataset.get_subset(Subset.VALIDATION)
+        config = self._config
+
+        time_monitor = TimeMonitorCallback(0, 0, 0, 0, update_progress_callback=lambda _: None)
```

Comment:
could you please implement this functionality and add corresponding tests, examples can be found in test_ote_api.py. I mean update_progress_callback